drivers/console/efi_console.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief EFI console driver
+ *
+ * @details EFI console driver.
+ * Hooks into the printk and fputc (for printf) modules.
+ */
+
+#include <stdio.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+
+
+#ifndef CONFIG_UART_CONSOLE
```

Comment:
If you're putting an entire file under an #if, it's best to just filter it out of the build using cmake instead.